src/reference/asciidoc/mqtt.adoc

```diff
@@ -101,6 +101,54 @@ The changes are not retained beyond the life cycle of the application context; a
 
 Changing the topics while the adapter is stopped (or disconnected from the broker) will take effect the next time a connection is established.
 
+==== Configuring with Java Configuration
+
+The following Spring Boot application provides an example of configuring the inbound adapter using Java configuration:
+[source, java]
+----
+@SpringBootApplication
+public class MqttJavaApplication {
+
+    public static void main(String[] args) {
+    	new SpringApplicationBuilder(MqttJavaApplication.class)
+    			.web(false)
+    			.run(args);
+    }
+
+    @Bean
+    public MessageChannel mqttInputChannel() {
+    	return new DirectChannel();
+    }
+
+    @Bean
+    public MessageProducer inbound() {
+    	MqttPahoMessageDrivenChannelAdapter adapter =
+    			new MqttPahoMessageDrivenChannelAdapter("tcp://localhost:1883", "testClient",
+    			                                 "topic1", "topic2");
+    	adapter.setCompletionTimeout(5000);
+    	adapter.setConverter(new DefaultPahoMessageConverter());
+    	adapter.setQos(1);
+    	adapter.setOutputChannel(mqttInputChannel());
+    	return adapter;
+    }
+
+    @Bean
+    @ServiceActivator(inputChannel = "amqpInputChannel")
```

Comment:
No, sorry.
Copy/paste legacy :smile:
Will appreciate if you fix on merge.
If it is only concern, of course...